examples/consumer_example/consumer_example.go

```diff
@@ -74,6 +74,9 @@ func main() {
 			case *kafka.Message:
 				fmt.Printf("%% Message on %s:\n%s\n",
 					e.TopicPartition, string(e.Value))
+				if e.Headers != nil {
```

Comment:
In C#, I currently return an empty collection rather than null when there are no headers, which has the advantage that it's a bit more convenient to use (null checks are not necessary) and consistency with Java (IIRC). downside is obviously memory usage / perf when there are no headers.

Reply:
Noted.
I think nil checks are pretty typical for Go though, and collection methods, such as len(), work on nil slices as well.

Reply:
sounds good to me